test/graphql/resolvers/volunteer_resolver_test.rb

```diff
@@ -0,0 +1,105 @@
+require_relative '../../test_helper'
+
+SingleCov.covered!
+
+describe VolunteerResolver do
+  let(:user) { users(:admin) }
+  let(:user2) { users(:a) }
+  let(:office) { offices(:remote) }
+  let(:office2) { offices(:madison) }
+
+  before do
+    Signup.delete_all
+    IndividualEvent.delete_all
+    user.update(office: office, role: Role.admin)
+  end
+
+  describe '.all' do
+    let(:event1) { events(:minimum) }
+    let(:event2) { events(:simple_event) }
+    let(:event3) { events(:late_event) }
+
+    it 'limits with time bounds' do
+      event1.update(starts_at: 2.weeks.ago, ends_at: (2.weeks.ago + 120.minutes))
+      event2.update(starts_at: 4.days.ago, ends_at: (4.days.ago + 20.minutes))
+      event3.update(starts_at: 3.days.ago, ends_at: (3.days.ago + 30.minutes))
+
+      Signup.delete_all
+      Signup.create!(user: user, event: event1)
+      Signup.create!(user: user, event: event2)
+      Signup.create!(user: user2, event: event3)
+
+      args = { after: 1.week.ago.to_i, before: 2.days.ago.to_i, sort_by: 'HOURS_DESC' }
+      results = VolunteerResolver.all(nil, args, nil).to_a
+
+      results.must_equal [user2, user]
+    end
+
+    it 'filters by all offices' do
+      args = { office_id: 'all' }
+
+      results = VolunteerResolver.all(nil, args, nil).to_a
+
+      results.must_equal User.all
+    end
+
+    it 'filters by current office' do
+      args = { office_id: 'current' }
+      context = { current_user: user }
+
+      results = VolunteerResolver.all(nil, args, context).to_a
+
+      results.must_equal [user]
+    end
+
+    it 'filters by office_id' do
+      args = { office_id: office.id }
+      results = VolunteerResolver.all(nil, args, nil).to_a
+
+      results.must_equal [user]
+    end
```

Comment:
These tests are failing. I believe there is some setup being done somewhere else for `UserResolver` which we have to replicate for `VolunteerResolver`.

Reply:
Actually, scrap the above, I just figured out why these don't work.

`UserResolver` does not rely on the user being signed up to an event to return a list of users, while `VolunteerResolver` does. This resolver was created to solve the problem that the top volunteer dashboard returns an incorrect list of top volunteers (hence the previous name). So it makes no sense to query for users without events.